src/pages/LVL_LevelEdit.page

```diff
@@ -13,125 +13,131 @@
-                            </div>
-                            <div class="slds-form-element slds-size_1-of-2">
-                                <apex:outputLabel for="fldMaxAmount" styleClass="slds-form-element__label uiLabel-left form-element__label uiLabel" >
-                                    {!$ObjectType.Level__c.Fields.Maximum_Amount__c.Label}
-                                </apex:outputLabel>
-                                <apex:inputField id="fldMaxAmount" required="false" value="{!Level__c.Maximum_Amount__c}" label="" styleClass="slds-input" />
+                                    </apex:inputField>
+                                </div>
+                                <div class="slds-form-element slds-size_1-of-2">
+                                    <apex:outputLabel for="fldMaxAmount" styleClass="slds-form-element__label uiLabel-left form-element__label uiLabel" >
+                                        {!$ObjectType.Level__c.Fields.Maximum_Amount__c.Label}
+                                    </apex:outputLabel>
+                                    <apex:inputField id="fldMaxAmount" required="false" value="{!Level__c.Maximum_Amount__c}" label="" styleClass="slds-input" />
+                                </div>
                             </div>
-                        </div>
 
-                        <div class="slds-form-element__row">
-                            <div class="slds-form-element slds-size_1-of-2">
-                                <apex:outputLabel for="fldSourceField" styleClass="slds-form-element__label uiLabel-left form-element__label uiLabel" >
-                                    <span class="required slds-required">*</span>{!$ObjectType.Level__c.Fields.Source_Field__c.Label}
-                                </apex:outputLabel>
-                                <apex:selectList id="fldSourceField" value="{!Level__c.Source_Field__c}" required="false" size="1" styleClass="slds-select">
-                                    <apex:selectOptions value="{!listSOSourceFields}"  />
-                                </apex:selectList>
+                            <div class="slds-form-element__row">
+                                <div class="slds-form-element slds-size_1-of-2">
+                                    <apex:outputLabel for="fldSourceField" styleClass="slds-form-element__label uiLabel-left form-element__label uiLabel" >
+                                        <span class="required slds-required">*</span>{!$ObjectType.Level__c.Fields.Source_Field__c.Label}
+                                    </apex:outputLabel>
+                                    <apex:selectList id="fldSourceField" value="{!Level__c.Source_Field__c}"
+                                                     required="false" size="1" styleClass="slds-select"
+                                                     disabled="{!NOT($ObjectType.Level__c.fields.Source_Field__c.updateable)}">
```

Comment:
As you noted in the WI, Source_Field__c and Level_Field__c are required fields, so it isn't possible for disabled to ever evaluate to true.  I'm wondering if it might make sense here to test whether Previous_Field__c is updateable.  I know that isn't what the bug bounty description requires, but it makes no sense to me that both Source_Field__c and Level_Field__c would be updateable while Previous_Field__c would not be.  Would it make the page more secure in a way that makes sense to use Previous_Field__c as a proxy for all 3?  If Previous_Field__c is not updateable, disable all 3.

Reply:
This really should change I think.  As the code currently stands, it is possible to leave a level__c record in an inconsistent state by changing source and level.  That leaves previous level potentially pointing to the wrong sobject.

Reply:
Do you mean something like the following:
- An admin creates a Level pointing to the Account object and sets the Level and Previous Level fields pointing to fields on the Account object
- A non-admin user (without edit rights on the Previous Level field) edits the Level to change the target object to Contact and selects a new target field for the Level field, leaving the PreviousLevel field with it's previous value.

Even if unlikely, it's not impossible. I think you're suggesting that if the User cannot edit PreviousLevel field that they should not be able to edit the SourceField field either? Effectively, the User would need edit access on all 3 fields to edit any of those 3 fields. I like that. Great suggestion. Change forthcoming (after I reboot).

Reply:
Yep.  That's the scenario I had in mind.  Definitely an obscure possibility, but I think your changes make the entire page more robust and secure.  Nice!